rfcs/text/0008-threat-intel.md

```diff
@@ -75,7 +81,178 @@ Stage 3: Add or update all remaining field definitions. The list should now be e
+    "event.module": "threatintel",
+
+    // The top-level file object here allows expressing multiple indicators for a single file object
+    "file.hash.sha256": "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4",
+    "file.hash.md5": "1eee2bf3f56d8abed72da2bc523e7431",
+    "file.size": 656896,
+    "file.name": "invoice.doc",
+
+    // The ioc prefix here gives context of the indicators
+    "ioc.marking.tlp": "WHITE",
+    "ioc.time_first_seen": "2020-10-01",
+    "ioc.time_last_seen": "2020-11-01",
+    "ioc.sightings": "4",
+    /* It's possible to have multiple related IOCs in a given document,
+       e.g. sha256, sha1, ssdeep, etc. If that's the case this should be an array
+       of types (i.e. [sha1, sha256, ssdeep]) */
+    "ioc.type": ["sha256", "md5", "file_name", "file_size"],
+    "ioc.description": "file last associated with delivering Angler EK",
+
+    // These would provide attribution-related information for the IOCs
+    "malware": {
+        "name": "CryptoMinerX.B",
+        "family": "CryptoMinerX",
+        "type": "cryptominer"
+    },
+    "threat_actor": {
+        "name": "CryptoCurrency R Us",
+        "type": [
+            "criminal"
+        ]
+    },
+
+    // Filebeats and other fields, not part of ECS proposal
+    "fileset.name": "abusemalware",
+    "input.type": "log",
+    "log.offset": 0,
+
+    // Any indicators should also be copied to relevant related.* field
+    "related": {
+        "hash": [
+            "1eee2bf3f56d8abed72da2bc523e7431",
+            "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4"
+        ]
+    },
+    "tags": [
+        "threatintel",
+        "forwarded"
+    ],
+}
+```
+
+2. **Adding threat intelligence match/enrichment to another document which could be in a source event index or signals index.** 
+
+    The Indicator Match Rule will be used to generate signals when a match occurs between a source event and threat intelligence document. The ECS fields proposed here will be used to add the enrichment and threat intel context in the signal document. 
+
+**Example**
+```json5
+{
+    "process": {
+        "name": "svchost.exe",
+        "pid": 1644,
+        "entity_id": "MDgyOWFiYTYtMzRkYi1kZTM2LTFkNDItMzBlYWM3NDVlOTgwLTE2NDQtMTMyNDk3MTA2OTcuNDc1OTExNTAw",
+        "executable": "C:\\Windows\\System32\\svchost.exe"
+    },
+    "message": "Endpoint file event",
+    "@timestamp": "2020-11-17T19:07:46.0956672Z",
+    "file": {
+        "path": "C:\\Windows\\Prefetch\\SVCHOST.EXE-AE7DB802.pf",
+        "extension": "pf",
+        "name": "SVCHOST.EXE-AE7DB802.pf",
+        "hash": {
+            "sha256": "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4"
+        }
+    },
+    "threat": [
+        // Each enrichment is added as a nested object under `threat.*`
+        {
+            "ioc": {
+                // Copy all the object IOCs under `ioc.*`, providing full context
+                "file": {
```

Comment:
Is `threat.ioc.file.*` another set of fields that should be captured in the proposed fields table?

Reply:
`threat.ioc.file.*` is the existing file fieldset. Any existing compatible IOC fieldset could be copied to `threat.ioc`, according to the matched rule. Happy to walk through this over a call and perhaps you can help us better clarify how it should be documented.